src/Enable.Extensions.Queuing.RabbitMQ/Internal/ConnectionManagement/RabbitMQManagedConnection.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using RabbitMQ.Client;
+
+namespace Enable.Extensions.Queuing.RabbitMQ.Internal
+{
+    internal class RabbitMQManagedConnection
+    {
+        public RabbitMQManagedConnection(IConnection connection)
+        {
+            Connection = connection;
+            Id = Guid.NewGuid();
+        }
+
+        public Guid Id { get; }
+
+        public IConnection Connection { get; private set; }
+
+        public int ReferenceCount { get; private set; } = 1;
+
+        public bool ChannelAllocationReached { get; set; }
+
+        public void DecrementReferenceCount()
+        {
+            ReferenceCount--;
```

Comment:
Let's use `Interlocked.Decrement` here. This ensure that this class is thread safe, and doesn't rely on calling code knowing that locks must be used with this method. The same applies to the `IncrementReferenceCount` method.

In other queue implementations, we return the value of the counter from these methods. For example, from #28:

```
 public int IncrementReferenceCount()
 {
    return Interlocked.Increment(ref _referenceCount
}
```

Can we do the same here for consistency (in fact, can we ensure that the RabbitMQ changes we're making here closely reflect what was done for the Service Bus changes)?

Reply:
This class is not intended to be thread safe. It is an internal class and only used in `RabbitMQConnectionManager` which is using locks when accessing these methods.  Changing to Interlocked adds a performance overhead with no benefit.